### DISCUSSION TITLE
I got some trouble deploy my contract on zksync

### DISCUSSION THREAD

**Discussion by Bytechoreographer** [Cyfrin/foundry-full-course-cu#2538] [Q&A]
According to the lesson 14, when I'm trying to deploy my contract on zksync sepolia test net, something went wrong. My contract is in a folder named contracts.
1. use my own wallet.
    I can't deploy my contract and there is a red exclamation. It seems that nothing happens and I can't see any error string, check and uncheck the verify box is the same.
![image](https://github.com/user-attachments/assets/90c5e27d-752e-4845-b8a7-3a875a92aae1)
2. use remote dev net
    It seems that I have successfully deployed the contract
![image](https://github.com/user-attachments/assets/cc4dc793-6210-48c4-a7b7-ad33c5d330d2)
But when I try to copy the address and search in [zksync ](https://sepolia.explorer.zksync.io/) like in the video, I can't see anything. There is no contract on the website. I find deploy in transaction tab. So how do I know if this contract is properly deployed. Here is the address in the picture: 0xf061E3b3D9d1e84c4cbF3AEb167f3C299e1BC09D
![image](https://github.com/user-attachments/assets/0252e022-e863-40b2-896d-d7b9ad95e202)
![image](https://github.com/user-attachments/assets/2ee0d55e-102f-4b16-b503-6cd4ccc1befa)
![image](https://github.com/user-attachments/assets/9c78c120-259f-4e1b-b529-75e9699795c7)

`{
  "type": 113,
  "nonce": 2763,
  "maxPriorityFeePerGas": {
    "type": "BigNumber",
    "hex": "0x0ee6b280"
  },
  "maxFeePerGas": {
    "type": "BigNumber",
    "hex": "0x0ee6b280"
  },
  "gasLimit": {
    "type": "BigNumber",
    "hex": "0x0359ae"
  },
  "to": "0x0000000000000000000000000000000000008006",
  "value": {
    "type": "BigNumber",
    "hex": "0x00"
  },
  "data": "0x9c4d535b0000000000000000000000000000000000000000000000000000000000000000010000d7bbc71b...",
  "chainId": {
    "type": "BigNumber",
    "hex": "0x010e"
  },
  "from": "0xa61464658AfeAf65CccaaFD3a512b69A83B77618",
  "customData": {
    "gasPerPubdata": {
      "type": "BigNumber",
      "hex": "0xc350"
    },
    "factoryDeps": "[ <...> ]",
    "customSignature": "0x541afc511db147bbcfb340f1d91e8894575f12ceb251dad1535b7f8104bc49203ae3548747182e3332f13b33772002ccb26f5558799dff2e15ef5c3c80081fea1c"
  },
  "hash": "0x670f03ff4b133459dfe0a74f88118800de5e6ed619247e90e52455fef24aafea",
  "confirmations": 0
}`

**Comment by mumernisar** [comment]
My two cents.. ! 
I don't know what devnet is since I did not deploy it with the tool but did it from the terminal later in the course! 
But certainly, you won't find it there. From what I can see, you have 999999 or some ether in that devnet wallet, and it's not easy to get that much ether! You have to use a bridge to get some ether from Sepolia test net into your Zksync Sepolia wallet!

**Reply by Bytechoreographer** [reply]
I have already bridged some ether to zksync sepolia wallet, as is shown in the first picture(I don't know why it is 0, maybe too small to show?）In the following pictures, you can see that I used remote devnet wallet
![image](https://github.com/user-attachments/assets/dc2d5864-b272-4f68-b215-e9ef4f0d5317)
![image](https://github.com/user-attachments/assets/9a1c0262-3abf-4d2c-8a7f-cf7542ceb18d)

**Comment by EngrPips** [comment]
Hello @Zhou-ZJ, Deploying to `DevNet` is like deploying to local Anvil chain, so you can't actually look up a contract you deployed to a local chain on a block explorer because it will not be recorded on there. Working with `ZKsync` can be difficult right now, and we expect that to get better with time. So if you are running into issues deploying to `ZkSync`, you can leave it and move on. Testing on Sepolia is sufficient, and that is why Patrick is always stressing it that `ZKSync` stuff is not compulsory

**Reply by Bytechoreographer** [reply]
Thanks a lot. I just followed the video and I'm not sure if I missed something. It works well in the video. I'll try it later.

**Reply by EngrPips** [reply]
Yeah, Let us know how it goes.
